samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -258,7 +290,9 @@ public void onBarrierCreated(String version) {
 
     public void onBarrierStateChanged(final String version, ZkBarrierForVersionUpgrade.State state) {
       LOG.info("JobModel version " + version + " obtained consensus successfully!");
+      metrics.barrierStateChange.inc();
       if (ZkBarrierForVersionUpgrade.State.DONE.equals(state)) {
+        metrics.singleBarrierRebalancingTime.update(System.nanoTime() - startTime);
```

Comment:
Sounds reasonable. Changed the timer metric to update after both state changes. Also fixed the startTime issue. Thanks!